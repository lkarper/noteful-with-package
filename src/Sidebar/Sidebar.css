.sidebar {
    width: 30%;
    display: flex;
    flex-direction: column;
}

.folder-link {
    border: 1px solid black;
    font-size: 18px;
    text-decoration: none;
    color: black;
    padding: 5px;
}

.highlighted {
    background: black;
    color: white;
    text-decoration: underline;
}

.folder-link:hover, .folder-link:focus {
    background-color: black;
}

.folder-link:hover , .folder-link:focus {
    color: white;
    text-decoration: underline;
}

.sidebar button {
    margin-top: 10px;
}

@media screen and (min-width: 750px) {
    .sidebar {
        width: 20%;
    }

    .folder-link {
        font-size: 25px;
    }
}