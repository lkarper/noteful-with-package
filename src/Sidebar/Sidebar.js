import React from 'react';
import { Route } from 'react-router-dom';
import SidebarLinks from '../SidebarLinks/SidebarLinks';
import NotesSidebar from '../NotesSidebar/NotesSidebar';
import SidebarError from '../SidebarError/SidebarError';
import './Sidebar.css';

const Sidebar = () => {
    
    return (
        <SidebarError>
            <Route 
                exact path={["/", "/folder/:folderId", "/add-folder", "/add-note"]}
                component={SidebarLinks}
            />
            <Route 
                path={["/note/:noteId", "/update-note/:noteId"]}
                component={NotesSidebar}
            />    
        </SidebarError>
    );
}

export default Sidebar;