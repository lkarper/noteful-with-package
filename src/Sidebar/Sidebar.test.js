import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Sidebar from './Sidebar';

describe.skip('Sidebar component', () => {
    it('renders the UI without crashing', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Sidebar />
            </BrowserRouter>
        );
        wrapper.unmount();
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Sidebar />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
    
});