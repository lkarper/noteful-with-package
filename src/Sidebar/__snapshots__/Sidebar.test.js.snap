// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Sidebar>
      <SidebarError>
        <Route
          component={[Function]}
          exact={true}
          path={
            Array [
              "/",
              "/folder/:folderId",
              "/add-folder",
              "/add-note",
            ]
          }
        >
          <SidebarLinks
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <nav
              className="sidebar"
            >
              <p>
                Loading...
              </p>
              <NavLink
                activeClassName="highlighted"
                className="folder-link"
                to="/add-folder"
              >
                <Link
                  aria-current={null}
                  className="folder-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/add-folder",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="folder-link"
                    href="/add-folder"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="folder-link"
                      href="/add-folder"
                      onClick={[Function]}
                    >
                      Add folder
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </nav>
          </SidebarLinks>
        </Route>
        <Route
          component={[Function]}
          path={
            Array [
              "/note/:noteId",
              "/update-note/:noteId",
            ]
          }
        />
      </SidebarError>
    </Sidebar>
  </Router>
</BrowserRouter>
`;
